src/main/java/ca/on/oicr/pineryreports/reports/impl/BisqueProjectsStatusReport.java

```diff
@@ -0,0 +1,769 @@
+    seqdLibsByCategory.get(COUNT_CATEGORY_NON_ILL_SEQD).put("All", new HashSet<>());
+
+    
+    // track which libraries where sequenced
+    for (RunDto run : allRuns) {
+      // ignore Running, Unknown, Stopped runs
+      if (!RUN_FAILED.equals(run.getState()) && !RUN_COMPLETED.equals(run.getState())) continue;
+      if (run.getPositions() == null) continue;
+      for (RunDtoPosition lane : run.getPositions()) {
+        if (lane.getSamples() == null) continue;
+        for (RunDtoSample sam : lane.getSamples()) {
+          // want only completed libraries for now
+          if (RUN_FAILED.equals(run.getState())) continue;
+          SampleDto dilution = allSamplesById.get(sam.getId());
+          if (isNonIlluminaLibrary(dilution)) {
+            seqdLibsByCategory.get(COUNT_CATEGORY_NON_ILL_SEQD).get("All").add(dilution);
+            continue;
+          }
+          String code = getUpstreamAttribute(ATTR_SOURCE_TEMPLATE_TYPE, dilution, allSamplesById);
+          if (code == null) throw new IllegalArgumentException("Dilution " + dilution.getName() + " has no library design code in hierarchy");
+          if (isRnaLibrary(dilution, allSamplesById)) {
+            if (is10XLibrary(dilution, allSamplesById)) {
+              // RNA, 10X
+              assignLibraryByTissueType(dilution, allSamplesById, seqdLibsByCategory.get(COUNT_CATEGORY_RNA_10X_SEQD));
+            } else {
+              // RNA, no 10X
+              assignLibraryByTissueType(dilution, allSamplesById, seqdLibsByCategory.get(COUNT_CATEGORY_RNA_SEQD));
+            }
+          } else {
+            if (is10XLibrary(dilution, allSamplesById)) {
+              // DNA, 10X
+              assignLibraryByTissueType(dilution, allSamplesById, seqdLibsByCategory.get(COUNT_CATEGORY_DNA_10X_SEQD));
+            } else if (LIBRARY_DESIGN_WG.equals(code)) {
+              // DNA, no 10X, WG
+              assignLibraryByTissueType(dilution, allSamplesById, seqdLibsByCategory.get(COUNT_CATEGORY_WG_SEQD));
+            } else if (LIBRARY_DESIGN_EX.equals(code)) {
+              // DNA, no 10X, EX
+              assignLibraryByTissueType(dilution, allSamplesById, seqdLibsByCategory.get(COUNT_CATEGORY_EX_SEQD));
+            } else if (LIBRARY_DESIGN_TS.equals(code)) {
+              // DNA, no 10X, TS
+              assignLibraryByTissueType(dilution, allSamplesById, seqdLibsByCategory.get(COUNT_CATEGORY_TS_SEQD));
+            } else if (LIBRARY_DESIGN_NN.equals(code)) {
+              // DNA, no 10X, NN
+              assignLibraryByTissueType(dilution, allSamplesById, seqdLibsByCategory.get(COUNT_CATEGORY_NN_SEQD));
+            } else if (ldO.contains(code)) {
+              // DNA, no 10X, AS/CH/BS
+              assignLibraryByTissueType(dilution, allSamplesById, seqdLibsByCategory.get(COUNT_CATEGORY_DNA_SEQD));
+            } else {
+              System.out.println("Unexpected library design code " + code + " was found on dilution " + dilution.getId());
```

Comment:
Might want to throw an exception here too, or you'll probably never check the log to see this?

Reply:
The 110 libraries that currently fall into this are Geospiza-old, and are WT/MR/TR libraries with DNA parents (might have been a migration error, might have been a data entry error). MISO keeps this particular error from happening, and we're not concerned about these 110 libraries being incorrect (they're on completed projects). I could either remove the check (as it should be impossible?) or I could create a list of "known errors" and only throw if a new library falls into this hole. I'm inclined towards the second, but am open to other suggestions!

Reply:
Hmm.. I see. Your original plan to log and ignore is probably best then, unless it would be useful to see a count of this bad data in the report?

Reply:
Or I could update the `isRnaLibrary` check to look at the library design code, instead of at whether the sample is DNA or RNA. That results in nothing logged and no errors at present, which sounds desirable to me; if we go ahead with this, maybe it should throw after all if it encounters this hole?